rust/src/lib.rs

```diff
@@ -2662,7 +2662,19 @@ pub unsafe extern "C" fn zcashlc_decrypt_and_store_transaction(
         //   from their encoding.
         let tx = Transaction::read(tx_bytes, BranchId::Sapling)?;
 
-        match decrypt_and_store_transaction(&network, &mut db_data, &tx) {
+        //
+        let mined_height = if mined_height > 0 {
+            let h = u32::try_from(mined_height)
+                .map_err(|e| anyhow!("Block height outside valid range: {}", e))?;
+            Some(h.into())
+        } else {
+            // We do not provide a mined height to `decrypt_and_store_transaction` for either
+            // transactions in the mempool or for transactions that have been mined on a fork
+            // but not in the main chain.
+            None
+        };
```

Comment:
Missing comment at the top.

Also, in the equivalent Android SDK change I handled this on the Kotlin side ([gRPC parsing](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/files#diff-af6f9221fb596338351cd262fe59664d5b2114ee07cabbd90de13864244991d5R33-R40), [conversion to main Kotlin type](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/files#diff-ccf0b549a7923796d7fd0266c843dc7264fa30f3a6b168c5106f0c9164021e00R30-R34)), and used `-1` separately as a sentinel for `None` across the FFI ([Kotlin side](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/files#diff-df1ff90ff1acc078d2f9983b61bc2bd21e6137845526b00c41d558df9358daccR315), [Rust side](https://github.com/Electric-Coin-Company/zcash-android-wallet-sdk/pull/1559/files#diff-63f3f4a8b98989699289a769aae68d9a8671559608f0119647df4a07194fcab3R1658)).

Reply:
Since we don't have a representation for the distinction between "in the mempool" and "otherwise mined in a fork" in `librustzcash` yet, I decided to allow the caller to directly propagate the response of the RPC call if they wanted; basically the only case we're currently interested in on the Rust side is when the mined height is known, all the other cases just get mapped to `None`. In the future if `librustzcash` begins to distinguish between these cases by using a bespoke enum instead of an `Option`, we'll update the FFI at that point.